### PR TITLE
Add `scale_info::TypeInfo` derives

### PR DESCRIPTION
**Pull request by ascjones**
Required for work on https://github.com/paritytech/substrate/discussions/8370, used in branch https://github.com/paritytech/substrate/compare/aj-metadata-vnext.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "finality-grandpa"
-version = "0.14.2"
+version = "0.15.0"
```
Comment (andresilva) [inline Cargo.toml:3]: Just bumping to `0.14.3` should suffice, there's no breaking change introduced here right?

**Review by andresilva — APPROVED** [review]
Changes LGTM, just one minor nit regarding version bump. Let me know when this should be merged (since I know the rest of the changes in substrate are still somewhat in flux).

**Comment by ascjones** [comment]
Happy for this to be merged/released now, so I can remove the `patch` for it in my substrate PR

**Comment by ascjones** [comment]
Ok for me to merge this now @andresilva?

**Pull request merged**